pkg/cfg/config_test.go

```diff
@@ -1,163 +1,259 @@
 package cfg_test
 
 import (
+	"fmt"
 	"github.com/applike/gosoline/pkg/cfg"
-	"github.com/stretchr/testify/assert"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/suite"
 	"math"
 	"strings"
 	"testing"
 	"time"
 )
 
-var baseSettings = map[string]interface{}{
-	"a": 1,
-	"n": map[string]interface{}{
-		"c": 3,
-	},
+type ConfigTestSuite struct {
+	suite.Suite
+
+	config      cfg.GosoConf
+	environment map[string]string
 }
 
-func TestConfig_AllKeys(t *testing.T) {
-	config := getNewTestableConfig(baseSettings, map[string]string{})
-	keys := config.AllKeys()
+func (s *ConfigTestSuite) SetupTest() {
+	s.environment = map[string]string{}
 
-	assert.Len(t, keys, 2)
+	options := []cfg.Option{
+		cfg.WithErrorHandlers(s.errorHandler),
+		cfg.WithEnvKeyReplacer(strings.NewReplacer(".", "_")),
+		cfg.WithSanitizers(cfg.TimeSanitizer),
+	}
+
+	s.config = cfg.NewWithInterfaces(s.environmentMock)
+
+	if err := s.config.Option(options...); err != nil {
```

Comment:
This seems duplicated from line 37, maybe we can just call that

Reply:
+1